bitchat/Features/voice/Waveform.swift

```diff
@@ -0,0 +1,95 @@
+import AVFoundation
+import Foundation
+import BitLogger
+
+/// Generates and caches downsampled waveforms for audio files so UI rendering is cheap.
+final class WaveformCache {
```

Comment:
I think a lot of these files dealing with custom queues could benefit from Swift Actors